homeassistant/components/freebox/sensor.py

```diff
@@ -177,6 +177,7 @@ def device_info(self) -> DeviceInfo:
     @callback
     def async_update_state(self) -> None:
         """Update the Freebox disk sensor."""
-        self._attr_native_value = round(
-            self._partition["free_bytes"] * 100 / self._partition["total_bytes"], 2
-        )
+        value = 0
+        if "total_bytes" in self._partition and self._partition["total_bytes"]:
```

Comment:
Why all these all extra checks? Just check if total_bytes == 0 and return 0 in that case.

Reply:
Yeah... Let's says it's "bad" habit inspired by the fact that some other parts of the code are failing due to inconsistent data.
Eg: this code in router.py is failing when a hard drive has no temperature sensor
`        for sensor in syst_datas["sensors"]:
            self.sensors_temperature[sensor["name"]] = sensor["value"]
`